pinot-core/src/main/java/com/linkedin/pinot/core/io/writer/impl/DirectMemoryManager.java

```diff
@@ -47,7 +48,7 @@ public DirectMemoryManager(final String segmentName) {
    */
   @Override
   protected PinotDataBuffer allocateInternal(long size, String allocationContext) {
-    return PinotDataBuffer.allocateDirect(size, allocationContext);
+    return PinotDataBuffer.allocateDirect(size, ByteOrder.nativeOrder(), allocationContext);
```

Comment:
This is using native order, and I see FixedByteSingleValueMultiColWriter is using BIG_ENDIAN. Lets document exactly how the byte ordering is handled (in all places in the code where applicable).

Reply:
Added comments